### PR TITLE
[Ready to Review] Feature/organize api tests

### PR DESCRIPTION
**Pull request by pattisdr**
# Purpose
Issue https://openscience.atlassian.net/browse/OSF-4444

Currently, all views tests for a given resource are in the same file, e.g. tests/api_tests/nodes/test_views.py.
These test files will quickly become hard to navigate as we add more endpoints (each endpoint has multiple test classes, e.g. TestNodeListCreate, TestNodeListFilter, ...).

# Changes

- Reorganized Nodes, Users, Files, Applications
Test files broken down so we have one file per endpoint
- e.g.
tests/api_tests/nodes/views/test_node_list.py (with TestNodeListCreate, TestNodeListFilter, etc.)
tests/api_tests/nodes/views/test_node_children.py

### CODE REVIEW

**Comment by sloria** [comment]
It doesn't appear that the api tests are being run with `invoke test` because the Travis build is failing, but when I run the API tests in isolation with `py.test tests/api_tests/` I get a number of expected errors:

```
tests/api_tests/nodes/views/test_node_files_list.py:269: in setUp
    {'name': 'xyz', 'path': '/xyz', 'materialized': '/xyz', 'kind': 'file'},
tests/api_tests/nodes/views/test_node_files_list.py:42: in prepare_mock_wb_response
    wb_url = waterbutler_api_url_for(node._id, provider=provider, path=path, meta=True)
E   NameError: global name 'waterbutler_api_url_for' is not defined
```

It's possible that the directories need to begin with `test_`, e.g. `test_views/` rather than `views/`, though I haven't looked into how nose's test discovery works. @pattisdr Can you look into this?

Also, can you please get this up to date with develop?

Once those are done, I will continue 1st pass.

**Comment by pattisdr** [comment]
Yes I will look into this.

**Comment by pattisdr** [comment]
I didn't have __init__.py in the folders, and then the new folders were just directories and not packages.  Since the tests were nested a couple of levels under these directories, since they didn't look like tests, they weren't inspected.  I changed the directories to packages so they would be inspected.

File: tests/api_tests/nodes/views/test_root.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
```
Comment (sloria) [inline tests/api_tests/nodes/views/test_root.py]: This file probably belongs in `tests/api_tests/base`.

**Comment by sloria** [comment]
Just one suggested change. Otherwise, this is really good.

**Comment by felliott** [comment]
Hey @pattisdr, could you bring this up-to-date with `develop` again?  There are a bunch of conflicts.  Once you do, I'll CR it immediately.  @sloria, if it passes my CR could we fast track it for merge?  Otherwise Dawn is going to have to keep unconflicting it everytime a new API branch with tests gets merged.

**Comment by sloria** [comment]
@felliott Yes, would like to get this in ASAP. Could you take a look at the file/folder structure? Updating to develop will not change the overall structure. If you're happy with this, I can take care of the conflicts when I merge.

**Comment by felliott** [comment]
@sloria, oh, cool!  I’ll start on it now.

**Comment by felliott** [comment]
CR2 passed.  Looks good to me.  I’m moving it to Ready to Merge.

**Pull request merged**